### PR TITLE
Update language around adapters, Apache Spark / Databricks

### PR DESCRIPTION
**Pull request by jtcohen6**
## Description & motivation

- The word **database** is intuitive for docs reader, but it's a little bit limiting and a little bit political. We're going to be adding more and more things to this page that are not databases per se. By contrast, although the word **adapter** feels [somewhat technical](https://en.wikipedia.org/wiki/Adapter_pattern), it most accurately describes the things that this page represents. I really believe that someone will know what this page means by recognizing at least one of the things listed. I haven't been able to come up with a great word or phrase that adequately sums up all of the following:
    - Postgres is a "relational database management system"
    - Redshift is a "managed ... data warehouse service"
    - Snowflake is a "data platform built for the cloud"
    - BigQuery is a "fully-managed, serverless data warehouse"
    - Apache Spark is a "unified analytics engine"
    - Databricks is a "Unified Data Analytics Platform"
    - Presto is a "distributed SQL query engine"
    - Dremio is a "cloud data lake engine"
    - ...
- Apache Spark is the full trademark. I imagine it's okay to shorten to "Spark" in a subsequent sentence (section?), but I'm taking the better-safe-than-sorry approach.
- Add Databricks to the list of Supported Adapters. It uses the same adapter plugin as Apache Spark, so it links to dedicated subheadings on the same pages.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [x] The page has been removed from `website/sidebars.js`
- [x] An entry has been added to `_redirects`

### CODE REVIEW

File: website/docs/docs/supported-adapters.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Supported adapters"
```
Comment (clrcrl) [inline website/docs/docs/supported-adapters.md]: As per Slack, I think "Available adapters" feels like the "correct" heading here.

That being said, I'm also curious what the most common search term is here — I think "supported databases" probably made a ton of sense for SEO purposes, but "available adapters" won't. Hmmmmm

Comment (jtcohen6) [reply website/docs/docs/supported-adapters.md]: Fair, I don't think it's a great step SEO-wise. All in all I'm not too worried about it:
- Searching for a specific supported adapter name ("Redshift," "Presto") still returns useful results
- [This FAQ](https://docs.getdbt.com/docs/about/overview/#what-databases-does-dbt-currently-support) still appears with search terms "database" + "support," and links to the right page (even though the page it's on is unlisted?)
- And I like that this page will now come up with the search term "available"

**Comment by clrcrl** [comment]
Heads up: you can `make build` locally to reproduce these errors :)

**Comment by jtcohen6** [comment]
I've definitely been doing `make run` locally instead... thank you!

**Review by clrcrl — APPROVED** [review]
LGTM!

**Pull request merged**